File: BabylonCS.JS/lib/scene.ts
import { Engine, Scene } from "@babylonjs/core";

export const createScene = (engine: Engine) => new Scene(engine);

export const sceneCreateDefaultCameraOrLight = (
  scene: Scene,
  createArcRotateCamera: boolean,
  replace: boolean,
  attachCameraControl: boolean
) =>
  scene.createDefaultCameraOrLight(
    createArcRotateCamera,
    replace,
    attachCameraControl
  );

export const sceneCreateDefaultEnvironment = (scene: Scene) =>
  scene.createDefaultEnvironment();

export const sceneRender = (scene: Scene) => scene.render();

export const sceneModule = {
  createScene,
  sceneCreateDefaultCameraOrLight,
  sceneCreateDefaultEnvironment,
  sceneRender,
};

File: BabylonCS/BabylonCS/Scene.cs
using System.Runtime.InteropServices.JavaScript;
using BabylonCS.Interop;

namespace BabylonCS;

public interface IScene
{
    void CreateDefaultCameraOrLight(bool createArcRotateCamera, bool replace, bool attachCameraControl);
    void Render();
}

public class Scene(Engine engine) : IScene
{
    internal JSObject SceneObj { get; } = SceneImpl.CreateScene(engine.EngineObj);

    public void CreateDefaultCameraOrLight(bool createArcRotateCamera, bool replace, bool attachCameraControl)
    {
        SceneImpl.SceneCreateDefaultCameraOrLight(SceneObj, createArcRotateCamera, replace, attachCameraControl);
    }

    public void Render()
    {
        SceneImpl.SceneRender(SceneObj);
    }
}